src/services/AuthService.js

```diff
@@ -0,0 +1,158 @@
+import jwtDecode from "jwt-decode";
+
+import httpService from "./HttpService";
+import { HTTP_METHODS, HTTP_STATUS_CODES } from "../consts";
+import { getItem, setItem, removeItem } from "../utils/localStorage";
+import config from "../config";
+
+const ROUTES = {
+  LOGIN: "/api/v1/token/",
+  ME: "/auth/me",
+  LOGOUT: "/auth/logout",
+  FORGOT_PASSWORD: "/user/forgot-password",
+  REGISTER: "/api/v1/users/",
+  RESET_PASSWORD: "/user/reset-password",
+  TOKEN_REFRESH: "/auth/refresh",
+};
+
+class AuthService {
+  constructor(httpService) {
+    this.httpService = httpService;
+  }
+
+  getAccessToken = () => {
+    return getItem("token");
+  };
+
+  setAuthToken = (token) => {
+    if (token) {
+      setItem("token", token);
+
+      this.httpService.attachHeaders({
+        Authorization: `Bearer ${token}`,
+      });
+    }
+  };
+
+  refreshToken = async (prevToken) => {
+    const { accessToken: newToken } = await this.httpService.request({
+      url: ROUTES.TOKEN_REFRESH,
+      method: HTTP_METHODS.POST,
+      headers: {
+        Authorization: `Bearer ${prevToken}`,
+      },
+    });
+
+    this.setAuthToken(newToken);
+
+    return newToken;
+  };
+
+  login = async (data) => {
+    data.username = data.email;
+    delete data.email;
```

Comment:
You can also override default username field (example), but for now you can leave this modification on front-end. 

Also, I would not modify data parameter directly, I would create something like:

const loginData = {
    username: data.email,
    password: data.password,
}